test/integration/authz_v2_test.go

```diff
@@ -0,0 +1,457 @@
+	_, err := framework.Global.IstioClientset.Update(updatedValue)
+	assert.Nil(t, err, "Update to Authorization policy must not fail")
+
+	// Check that the original authorization policy is restored.
+	rolloutAndValidateAuthorizationPolicyScenario(t, e, noop, noop)
+	cleanupAuthorizationRbac(t, e)
+}
+
+// Manual delete AP
+// Initial: Existing service, AP, AD
+// Input Actions: delete AP resource
+// Output: AP recreated to match with existing AD
+func TestDeleteAuthorizationPolicyRestoresOriginal(t *testing.T) {
+	// Initial set up
+	e := fixtures.GetBasicRbacV2Case(nil)
+	rolloutAndValidateAuthorizationPolicyScenario(t, e, create, create)
+
+	// Delete the Authorization policy created
+	ap := e.AuthorizationPolicies[0]
+	err := framework.Global.IstioClientset.Delete(ap.GroupVersionKind(), ap.Name, ap.Namespace)
+	assert.Nil(t, err, "Delete of authorization policy should not fail")
+
+	// Verification and cleanup
+	rolloutAndValidateAuthorizationPolicyScenario(t, e, noop, noop)
+	cleanupAuthorizationRbac(t, e)
+}
+
+// Update AD
+// Initial: Existing service with annotation, AP
+// Input Actions: update athenz domain with service related role, or policies associated to the role
+// Output: AP updated
+func TestUpdateAthenzDomainUpdatesAuthorizationPolicy(t *testing.T) {
+	// Initial set up
+	e := fixtures.GetBasicRbacV2Case(nil)
+	rolloutAndValidateAuthorizationPolicyScenario(t, e, create, create)
+
+	// Get updated fixtures
+	newUserName := "user.bar"
+	modified := fixtures.GetBasicRbacV2Case(&fixtures.RbacV2Modifications{
+		ModifyAthenzDomain: []func(signedDomain *zms.SignedDomain){
+			func(signedDomain *zms.SignedDomain) {
+				signedDomain.Domain.Roles[0].Members = append(signedDomain.Domain.Roles[0].Members, zms.MemberName(newUserName))
+				signedDomain.Domain.Roles[0].RoleMembers = append(signedDomain.Domain.Roles[0].RoleMembers, &zms.RoleMember{MemberName: zms.MemberName(newUserName)})
+			},
+		},
+		ModifyAuthorizationPolicies: [][]func(policy *v1beta1.AuthorizationPolicy){
+			[]func(policy *v1beta1.AuthorizationPolicy){
+				func(policy *v1beta1.AuthorizationPolicy) {
+					policy.Rules[0].From[0].Source.Principals = append(policy.Rules[0].From[0].Source.Principals, "user/sa/bar")
+					principals := policy.Rules[0].From[0].Source.Principals
+					principals[1], principals[2] = principals[2], principals[1]
+					requestPrincipals := []string{policy.Rules[0].From[1].Source.RequestPrincipals[0], "athenz/user.bar"}
+					policy.Rules[0].From[1].Source.RequestPrincipals = requestPrincipals
+				},
+			},
+		},
+	})
+
+	rolloutAndValidateAuthorizationPolicyScenario(t, modified, update, noop)
+	cleanupAuthorizationRbac(t, modified)
+}
+
+// Update AD
+// Initial: Existing service with annotation, AP
+// Input Actions: update athenz domain with service related role, or policies associated to the role
+// Output: AP updated
+func TestUpdateAthenzDomainIgnoresAuthorizationPolicyWithOverrideAnnotation(t *testing.T) {
```

Comment:
I think the test case should be: existing ap with override annotation set, input actions: update ap manually with policy addition, output: ap is update(does not sync with ad)
lmk if this makes sense

Reply:
Yup, makes sense. Updated the comment with the suggested text.

Reply:
thank you